modules/Sources/Features/ImportWallet/ImportWallet.swift

```diff
@@ -60,31 +61,52 @@ public struct ImportWallet: Reducer {
     }
     
     public var body: some ReducerOf<ImportWallet> {
+        BindingReducer()
+        
         Reduce { state, action in
             switch action {
             case .alert(.dismiss):
                 return .none
-            case .birthdayInputChanged(let redactedBirthday):
+            case .binding(\.$birthdayHeight):
                 let saplingActivation = saplingActivationHeight
-
-                state.birthdayHeight = redactedBirthday
-
-                if let birthdayHeight = BlockHeight(state.birthdayHeight.data), birthdayHeight >= saplingActivation {
+                if let birthdayHeight = BlockHeight(state.birthdayHeight), birthdayHeight >= saplingActivation {
                     state.birthdayHeightValue = birthdayHeight.redacted
                 } else {
                     state.birthdayHeightValue = nil
                 }
                 return .none
+            case .binding(\.$importedSeedPhrase):
+                let trimmedPhraseWords = state.importedSeedPhrase.split(separator: " ")
+                    .filter { !$0.isEmpty }
+                    .map { $0.trimmingCharacters(in: .whitespaces) }
+                state.wordsCount = trimmedPhraseWords.count
+                state.isValidNumberOfWords = state.wordsCount == state.maxWordsCount
+                // is the mnemonic valid one?
+                do {
+                    try mnemonic.isValid(trimmedPhraseWords.joined(separator: " "))
+                } catch {
+                    state.isValidMnemonic = false
+                    return .none
+                }
+                state.isValidMnemonic = true
+                return .none
+            case .binding:
+                return .none
             case .continueTapped:
                 do {
+                    let trimmedPhraseWords = state.importedSeedPhrase.split(separator: " ")
+                        .filter { !$0.isEmpty }
+                        .map { $0.trimmingCharacters(in: .whitespaces) }
+                    
                     // validate the seed
-                    try mnemonic.isValid(state.importedSeedPhrase.data)
+                    let cleanedPhrase = trimmedPhraseWords.joined(separator: " ")
+                    try mnemonic.isValid(cleanedPhrase)
```

Comment:
Why is the cleaning and validation performed again, is there a way of continuing when the form is invalid?

Reply:
This logic was adjusted to make entering seed phrase a bit more forgiving, particularly around whitespace errors. Previously, if a user entered any extra whitespace between words, this would be considered an error and result in validation failing. This seemed extreme for an issue that could reasonably be compensated for by removing extra whitespace. If this is a security concern, I can revert it. Just seemed like a simple UX enhancement to make.